crossdock/crossdock_test.go

```diff
@@ -123,12 +113,45 @@ func TestCrossdock(t *testing.T) {
 	}
 }
 
+func wait(t *testing.T) {
+	attempts := 0
+	for {
+		if attempts > 9 {
+			t.Fatalf("could not talk to client in %d attempts", attempts)
+		}
+
+		attempts++
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
```

Comment:
From [context's docs](https://godoc.org/golang.org/x/net/context#WithTimeout):

> Canceling this context releases resources associated with it, so code should call cancel as soon as the operations running in this Context complete:

```go
func slowOperationWithTimeout(ctx context.Context) (Result, error) {
	ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
	defer cancel()  // releases resources if slowOperation completes before timeout elapses
	return slowOperation(ctx)
}
```

This makes me feel like I'm doing it right.